builtin/logical/database/credentials_test.go

```diff
@@ -17,6 +17,262 @@ import (
+				KeyBits: 3072,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with 4096 value key_bits",
+			args: args{
+				config: map[string]interface{}{
+					"key_bits": "4096",
+				},
+			},
+			want: ClientCertificateGenerator{
+				KeyType: "ec",
+				KeyBits: 4096,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with 224 value key_bits",
+			args: args{
+				config: map[string]interface{}{
+					"key_bits": "224",
+				},
+			},
+			want: ClientCertificateGenerator{
+				KeyType: "ec",
+				KeyBits: 224,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with 256 value key_bits",
+			args: args{
+				config: map[string]interface{}{
+					"key_bits": "256",
+				},
+			},
+			want: ClientCertificateGenerator{
+				KeyType: "ec",
+				KeyBits: 256,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with 384 value key_bits",
+			args: args{
+				config: map[string]interface{}{
+					"key_bits": "384",
+				},
+			},
+			want: ClientCertificateGenerator{
+				KeyType: "ec",
+				KeyBits: 384,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with 521 value key_bits",
+			args: args{
+				config: map[string]interface{}{
+					"key_bits": "521",
+				},
+			},
+			want: ClientCertificateGenerator{
+				KeyType: "ec",
+				KeyBits: 521,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with invalid key_bits",
+			args: args{
+				config: map[string]interface{}{
+					"key_bits": "4097",
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "newClientCertificateGenerator with zero value signature_bits",
+			args: args{
+				config: map[string]interface{}{
+					"signature_bits": "0",
+				},
+			},
+			want: ClientCertificateGenerator{
+				KeyType:       "ec",
+				SignatureBits: 0,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with 256 value signature_bits",
+			args: args{
+				config: map[string]interface{}{
+					"signature_bits": "256",
+				},
+			},
+			want: ClientCertificateGenerator{
+				KeyType:       "ec",
+				SignatureBits: 256,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with 384 value signature_bits",
+			args: args{
+				config: map[string]interface{}{
+					"signature_bits": "384",
+				},
+			},
+			want: ClientCertificateGenerator{
+				KeyType:       "ec",
+				SignatureBits: 384,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with 512 value signature_bits",
+			args: args{
+				config: map[string]interface{}{
+					"signature_bits": "512",
+				},
+			},
+			want: ClientCertificateGenerator{
+				KeyType:       "ec",
+				SignatureBits: 512,
+			},
+		},
+		{
+			name: "newClientCertificateGenerator with invalid signature_bits",
+			args: args{
+				config: map[string]interface{}{
+					"signature_bits": "112",
+				},
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newClientCertificateGenerator(tt.args.config)
+			if tt.wantErr {
+				assert.Error(t, err)
```

Comment:
nit: It would be good to specific about the exact error instead of just the presence of an error.